### PR TITLE
Fixing tests to not depend on background thread for cleanup verification

### PR DESCRIPTION
**Pull request by kichalla**
@Tratcher what do you think?

### CODE REVIEW

**Comment by Tratcher** [comment]
:cry:

**Comment by Tratcher** [comment]
If it's got to be hacked, at least abstract it out like ISystemClock. That said, I'd prefer not to modify the MemoryCache, just the tests.

File: src/Microsoft.Framework.Caching.Memory/CacheEntry.cs [hunk 1]
```diff
@@ -39,6 +39,9 @@ internal class CacheEntry
 
         internal object Value { get; private set; }
 
+        // To enable unit testing
+        internal bool RunCleanupInBackgroundThread { get; set; } = true;
```
Comment (davidfowl) [inline src/Microsoft.Framework.Caching.Memory/CacheEntry.cs]: Zomg

**Comment by muratg** [comment]
@kichalla What issue does this one fix?  And are we taking it?

**Comment by muratg** [comment]
@kichalla

**Comment by kichalla** [comment]
@muratg I am taking a look at it now.

**Comment by kichalla** [comment]
Discussed with @Tratcher and closing this PR for now and instead make changes as per the following comment to start with: https://github.com/aspnet/Caching/issues/106#issuecomment-140143178

**Pull request closed without merging**